### ISSUE TITLE
Sorting by "Time Sent" is messed up

### ISSUE THREAD

**Issue by cjoergensen** [Particular/ServiceControl#1804]
We are seeing this behavior in ServiceInsight:

![sort-by-timesent-page1](https://user-images.githubusercontent.com/8087733/67682847-901df500-f990-11e9-8cf3-a931c60cda03.png)
![sort-by-timesent-page2](https://user-images.githubusercontent.com/8087733/67682856-92804f00-f990-11e9-9d09-2921127f8f54.png)

It seems to be a problem in ServiceControl, when inspecting the data returned from the api.

Last item int page 1: (/api/messages/?include_system_messages=False&page=0&per_page=50&sort=time_sent&direction=desc)

```
    "time_sent": "2019-10-16T15:32:47.710592Z",
    "processed_at": "2019-10-21T10:21:34.753333Z",
```

And first item from page 2: (/api/messages/?include_system_messages=False&page=1&per_page=50&sort=time_sent&direction=desc)

```
    "time_sent": "2019-10-28T13:24:11.494959Z",
    "processed_at": "2019-10-28T13:24:12.765037Z",
```

**Comment by danielmarbach** [comment]
@cjoergensen Are you running ServiceControl in multi-instance mode? Or did you recently upgrade from 3.x to 4.4.1 where we introduced the ServiceControl Audit instance?

**Comment by cjoergensen** [comment]
@danielmarbach We are not running in multi-instance mode, just a plain installation. 

I think we updated to from 3.x to 4.0.1 about two months ago and we have been regular updated it since then.

**Comment by danielmarbach** [comment]
@cjoergensen thanks for coming back to me.

We are aware of that inconsistency and we are considering introducing a more consistent sorting behavior. I can keep you up to date here once we know more.

From an audit message perspective, this behavior is a temporary one though that is caused by the primary instance having audit data (but no longer ingesting) and new audit data arriving in the new audit instance (created from the split in 4.x). Once the expiration policy kicks in the primary instance audit data will be removed and only the audit data in the new audit instance remains. Once that clean up is over the sorting behavior should be aligned again with the experience you probably expected.

**Comment by cjoergensen** [comment]
@danielmarbach i would appreciate if you keep me posted on updated. 

I should note that we have a very long retention policy (190 days!) on our audit messages.

**Comment by danielmarbach** [comment]
@cjoergensen I would love to hear why you have such a long retention policy to better understand what you are doing. Could you elaborate a bit? If you prefer to share it via email you can send an email to daniel dot marbach at particular dot net

**Comment by TraGicCode** [comment]
@danielmarbach 

I think we are runnining into this same issue. Is there something we can do in ravendb to expire the remaining audit messages from the primary servicecontrol instance?

**Comment by danielmarbach** [comment]
@TraGicCode sorry for the long silence period. 

Have you seen that you can control the data retention settings including the aggressiveness of expiring the data via the settings?

https://docs.particular.net/servicecontrol/creating-config-file#data-retention

In case you want to get rid of all of them you could set the main instance into maintenance mode and then delete the ProcessedMessage collection

![image](https://user-images.githubusercontent.com/174258/70121696-67fc7280-166f-11ea-88ec-b13af45c6a63.png)

![image](https://user-images.githubusercontent.com/174258/70121750-8d897c00-166f-11ea-9a52-dde2bcb61a45.png)

![image](https://user-images.githubusercontent.com/174258/70121772-9aa66b00-166f-11ea-9846-6a9979e17b74.png)

The problem is that it doesn't remove the attachments then so I would suggest sticking to tweak the data retention and expiry cleanup intervals to your need on the main instance.

**Comment by TraGicCode** [comment]
Hey @danielmarbach 

Are you saying on the primary service control instance i can change the audit retention to like 1 day and it should expire all of them by tommorow?  Also can you verify the exact appsettings key for the primary instance?  Is it

ServiceControl.Audit/AuditRetentionPeriod OR ServiceControl/AuditRetentionPeriod?  I would expect it to be ServiceControl/AuditRetentionPeriod if i had to guess

**Comment by danielmarbach** [comment]
> Are you saying on the primary service control instance i can change the audit retention to like 1 day and it should expire all of them by tommorow?

Yes but I would not give any guarantees in how long it takes until they are gone

It is

`ServiceControl/AuditRetentionPeriod`

**Comment by cjoergensen** [comment]
@danielmarbach sorry for the long response time. 

Based on your response we wen't back and discussed our retention policy. The 190 day policy was based on the need for some system to provide audit trails for legal reasons and it was very convenient to just use the build in audit feature in NSB. 

But that is really not what the SC is build for, so we did a bit of redesign and has now cut our retention period down to 90 days.

**Comment by boblangley** [comment]
We have triaged this issue and added it to our backlog. We will work on it as soon as we can but we do not have an ETA at this time. We will update this issue when work begins.
